shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/PatternsByStop.kt

```diff
@@ -102,6 +102,52 @@ data class PatternsByStop(
             .distinct()
     }
 
+    /**
+     * Returns a unique list of alerts downstream on the routes served by the target stop. For each
+     * route pattern in the given direction, includes only the alerts at the first downstream stop
+     * that has any alerts.
+     */
+    fun downstreamAlertsFor(directionId: Int, global: GlobalResponse): List<Alert> {
+
+        val patternsInDirection = this.patterns.filter { it.directionId() == directionId }
+        val stopIds = setOf(this.stop.id) + this.stop.childStopIds
+
+        return patternsInDirection
+            .flatMap { realtime ->
```

Comment:
I'm probably going to refactor at least a chunk of this logic into RealtimePatterns for reusability for surfacing downstream disruptions as a secondary alert in nearby transit

Reply:
Ugly WIP branch of that refactor: https://github.com/mbta/mobile_app/pull/new/kb-downstream-alerts-nearby

Reply:
Merged in